client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/InteractionSection/InteractionDetailsFields/InteractionTimeForm.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import {Controller, useFormContext} from 'react-hook-form';
+import {isValid} from 'date-fns';
+import {Checkbox, FormControlLabel, Grid} from '@material-ui/core';
+
+import InteractionEventDialogFields
+    from 'models/enums/InteractionsEventDialogContext/InteractionEventDialogFields';
+import {get} from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
+import FormInput from 'commons/FormInput/FormInput';
+import TimePick from 'commons/DatePick/TimePick';
+import useFormStyles from 'styles/formStyles';
+
+const TimeForm = () => {
+    const formClasses = useFormStyles();
+    const { control, watch, errors, setValue, clearErrors, setError } = useFormContext();
+    const startTime = watch(InteractionEventDialogFields.START_TIME) || null;
+    const endTime = watch(InteractionEventDialogFields.END_TIME) || null;
+    const isUnknownTime = watch(InteractionEventDialogFields.UNKNOWN_TIME);
+
+    const isEndTimeValid = (fieldName : string, currentTime : Date ) => {
+        if(fieldName === InteractionEventDialogFields.END_TIME) {
+            return startTime && currentTime.getTime() > startTime.getTime()
+        }
+        return true;
+    };
+
+    const handleTimeChange = (currentTime: Date, interactionDate: Date, fieldName: string) => {
+        if (isValid(currentTime)) {
+            if(isEndTimeValid(fieldName , currentTime)) {
+                let newDate = new Date(interactionDate.getTime());
+
+                newDate.setHours(currentTime.getHours());
+                newDate.setMinutes(currentTime.getMinutes());
+
+                if (newDate.getTime()) {
+                    clearErrors(fieldName);
+                    setValue(fieldName, newDate);
+                }
+            } else {
+                setError(fieldName, {type: 'manual', message: 'שעת סיום לא תקינה'});
+            }
+        } else {
+            setError(fieldName, {type: 'manual', message: 'שעה לא תקינה'});
+        }
+    };
+
+    return (
+        <Grid className={formClasses.formRow} container justify='flex-start'>
+            <FormInput xs={5} fieldName='משעה'>
+                <Controller
+                    name={InteractionEventDialogFields.START_TIME}
+                    control={control}
+                    render={(props) => (
+                        <TimePick
+                            disabled={isUnknownTime as boolean}
```

Comment:
here and in line 73 use Boolean(isUnknownTime) instead, think what happens when isUnknownTime is undefined , 
it'll be disabled={undefined} instead of false

Reply:
or even declate isUnknownTime as Boolean(watch(InteractionEventDialogFields.UNKNOWN_TIME))

Reply:
this is old code, just copied to separate component so i am unfamiliar with the behaviour

Reply:
but undefined is falsy in JS so i guess the expected behaviour would be to treat undefined as false which it does

Reply:
i may be misunderstanding the comment

Reply:
🤷‍♂️